src/core/execute.c

```diff
@@ -3852,16 +3842,12 @@ static int apply_mount_namespace(
                                        "Bind mounts: %zu, temporary filesystems: %zu, root directory: %s, root image: %s, dynamic user: %s",
                                        n_bind_mounts, context->n_temporary_filesystems, yes_no(root_dir), yes_no(root_image), yes_no(context->dynamic_user));
 
-                        r = -EOPNOTSUPP;
-                } else {
+                        return -EOPNOTSUPP;
+                } else
                         log_unit_debug(u, "Failed to set up namespace, assuming containerized execution and ignoring.");
-                        r = 0;
-                }
         }
 
-finalize:
-        bind_mount_free_many(bind_mounts, n_bind_mounts);
-        return r;
```

Comment:
Oh, all errors except for `-ENOANO` are ignored now.

Reply:
Oops.

Reply:
I think this should fix `TEST-43-PRIVATEUSER-UNPRIV`.